### PR TITLE
Final PR for Java Authenticator Guides

### PR DESCRIPTION
**Pull request by danielmaharry-okta**
## Description:
- **What's changed?** Final PR for the Java versions of five guides. All guides have been final edited already. These were previously blocked by an issue with the Device Context feature in Identity Engine and so were grouped together for quick release once unblocked. It has been agreed to push these live despite the DC issue remaining unresolved. Okta Verify for now is commented out in situ, with a warning in its place.

- **Is this PR related to a Monolith release?** No

### Resolves:

* [SSPR with Embedded Auth integration guide](https://oktainc.atlassian.net/browse/OKTA-459209)
* [Google authenticator integration guide](https://oktainc.atlassian.net/browse/OKTA-459226)
* [Okta Email integration guide](https://oktainc.atlassian.net/browse/OKTA-459283)
* [Okta Verify integration guide](https://oktainc.atlassian.net/browse/OKTA-459398)
* [WebAuthn integration guide](https://oktainc.atlassian.net/browse/OKTA-459404)

### CODE REVIEW

File: packages/@okta/vuepress-site/docs/guides/authenticators-google-authenticator/main/java/challengeintegrationsteps.md [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+```java
+case AWAITING_AUTHENTICATOR_SELECTION:
+    // Request the list of authenticators, and then present a selection list to the user.
+    // authenticationResponse.getAuthenticators()
+
+    // Continue with the selected factor.
+    authenticationResponse = idxAuthenticationWrapper.selectAuthenticator(proceedContext, authenticator);
+```
+
+> **NOTE:**  This state is skipped if your application's sign-on policy contains only one factor.
```
Comment (robertnicolo-okta) [inline packages/@okta/vuepress-site/docs/guides/authenticators-google-authenticator/main/java/challengeintegrationsteps.md:24]:
```suggestion
> **Note:**  This state is skipped if your application's sign-on policy contains only one factor.
```

File: packages/@okta/vuepress-site/docs/guides/authenticators-google-authenticator/main/java/challengeintegrationsteps.md [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+![A one-time password being shown in Google Authenticator](/img/authenticators/authenticators-google-one-time-password.png)
+
+</div>
+
+### 7 - 8: Display challenge page and process password
+
+The challenge flow now follows the same final steps as the [Enrollment flow](#_8-challenge-user-for-totp):
+
+* Challenge user for TOTP
+* Sign user in
```
Comment (robertnicolo-okta) [inline packages/@okta/vuepress-site/docs/guides/authenticators-google-authenticator/main/java/challengeintegrationsteps.md:41]: The See Also section is empty

File: packages/@okta/vuepress-site/docs/guides/oie-embedded-sdk-use-case-custom-pwd-recovery-mfa/main/java/siw.md [hunk 3]
```diff
@@ -0,0 +1,116 @@
...
+  </script>
+```
+
+### 3: Display password reset prompt and complete password recovery flow
+
+After the widget loads, it checks the validity of the `otp` and `state` values with the Okta server. If they are valid, the user is prompted to reset their password, as shown below. The user continues the password recovery flow described in the [User password recovery summary of steps](/docs/guides/oie-embedded-sdk-use-case-pwd-recovery-mfa/java/main/#summary-of-steps).
+
+<div class="common-image-format">
+
+![Screenshot of password reset page](/img/advanced-use-cases/java-custom-pwd-recovery-custom-siw-reset-pwd-page.png "Password Reset Page")
```
Comment (robertnicolo-okta) [inline packages/@okta/vuepress-site/docs/guides/oie-embedded-sdk-use-case-custom-pwd-recovery-mfa/main/java/siw.md:114]: screenshot missing border

**Review by robertnicolo-okta — CHANGES_REQUESTED** [review]
See each feedback item for more details.

File: packages/@okta/vuepress-site/docs/guides/authenticators-okta-email/main/java/integratechallengemagiclink.md [hunk 4]
```diff
@@ -0,0 +1,184 @@
...
+        … your code …
+}
+```
+
+If you configured your Okta org correctly, you need to respond to two specific authenticator statuses to handle this scenario:
+
+* `AWAITING_AUTHENTICATOR_SELECTION` that is covered in this section
+* `AWAITING_AUTHENTICATOR_VERIFICATION` that is covered in a later section
+
+### 4. Display a list of possible authenticators
```
[the remark this answers is not in the document]
Comment (danielmaharry-okta) [reply packages/@okta/vuepress-site/docs/guides/authenticators-okta-email/main/java/integratechallengemagiclink.md:58]: changed

File: packages/@okta/vuepress-site/docs/guides/authenticators-okta-email/main/java/integratechallengemagiclink.md [hunk 5]
```diff
@@ -0,0 +1,184 @@
...
+      // The emails also contain a TOTP code.
+      // You could implement both polling and the ability to enter a code.
+      sleep(Integer.parseInt(authenticationResponse
+         .getCurrentAuthenticatorEnrollment().getValue().getPoll().getRefresh()));
+
+      return idxAuthenticationWrapper.poll(proceedContext);
+    }
+
+    throw new IllegalStateException(
+      "Unsupported factor type selected: " + factorType);
```
[the remark this answers is not in the document]
Comment (danielmaharry-okta) [reply packages/@okta/vuepress-site/docs/guides/authenticators-okta-email/main/java/integratechallengemagiclink.md:120]: @robertnicolo-okta  I've deleted this section's text and replaced it with the text from the same area in the aspnet guide. This covers both options and does not mention polling.

File: packages/@okta/vuepress-site/docs/guides/authenticators-google-authenticator/main/java/challengeintegrationsteps.md [hunk 6]
```diff
@@ -0,0 +1,41 @@
+### 1 - 4: Start flow and display authenticator list
+
+The challenge flow follows the same first four steps as the [enrollment flow](#integrate-sdk-for-authenticator-enrollment):
```
[the remark this answers is not in the document]
Comment (danielmaharry-okta) [reply packages/@okta/vuepress-site/docs/guides/authenticators-google-authenticator/main/java/challengeintegrationsteps.md:3]: @robertnicolo-okta  We talked about this briefly already.

- If there are a number of steps which need to be documented but don't contribute to the actual flow, I compact them into one step. For example 1-3. Start flow and retrieve list of authenticators.  Steps 1 to 3 are written out in full in the text. 
- The only reason I'd abridge the text is if those steps are excatly the same as a flow covered earlier in the same document. For example in the Email authetnicator document I have "1 - 7. Sign in and get magic link email
First, the user attempts to sign in, initiating the email challenge flow. Next, they click the magic link that redirects them back to your app. These steps are identical to steps 1 - 7 in Integrate email challenge with magic links." 

I hope that's OK.

File: packages/@okta/vuepress-site/docs/guides/authenticators-google-authenticator/main/java/challengeintegrationsteps.md (hunk 1 above)
Comment (danielmaharry-okta) [reply packages/@okta/vuepress-site/docs/guides/authenticators-google-authenticator/main/java/challengeintegrationsteps.md:24]: fixed

File: packages/@okta/vuepress-site/docs/guides/authenticators-google-authenticator/main/java/challengeintegrationsteps.md (hunk 2 above)
Comment (danielmaharry-okta) [reply packages/@okta/vuepress-site/docs/guides/authenticators-google-authenticator/main/java/challengeintegrationsteps.md:41]: fixed.

File: packages/@okta/vuepress-site/docs/guides/oie-embedded-sdk-use-case-custom-pwd-recovery-mfa/main/java/siw.md (hunk 3 above)
Comment (danielmaharry-okta) [reply packages/@okta/vuepress-site/docs/guides/oie-embedded-sdk-use-case-custom-pwd-recovery-mfa/main/java/siw.md:114]: fixed

File: packages/@okta/vuepress-site/docs/guides/authenticators-google-authenticator/main/java/challengeintegrationsteps.md (hunk 6 above)
[the remark this answers is not in the document]
Comment (robertnicolo-okta) [reply packages/@okta/vuepress-site/docs/guides/authenticators-google-authenticator/main/java/challengeintegrationsteps.md:3]: There was one section (we just talked about) which you'll add links pointing to the other section. I'm going to LGTM to not hold you up.

**Review by robertnicolo-okta — APPROVED** [review]
lgtm

**Pull request merged**